Provide translation of the input page (#577)
reactjs/fr.react.dev: src/content/reference/react-dom/components/input.md

<Intro>

> Le [composant natif `<input>` du navigateur](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input) vous permet d'afficher différents champs de saisie dans un formulaire.

Reviewer: ```suggestion
Le [composant natif `<input>` du navigateur](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input) vous permet d'afficher différents types de champs de saisie dans un formulaire.
```

On n'affiche pas plusieurs champs d'un coup, légère nuance.

---

## Référence {/*reference*/}

### `<input>` {/*input*/}

Pour afficher un champ de saisie, utilisez le [composant natif `<input>` du navigateur](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input).

```js
<input name="myInput" />
```

[Voir plus d'exemples ci-dessous](#usage).

#### Props {/*props*/}

`<input>` prend en charge toutes les [props communes aux éléments](/reference/react-dom/components/common#props).

Un champ de saisie peut devenir un [champ contrôlé](#controlling-an-input-with-a-state-variable) en lui passant une de ces props :

> * [`checked`](https://developer.mozilla.org/fr/docs/Web/API/HTMLInputElement#checked): un booléen. Pour un champ de saisie avec une case à cocher ou un bouton de type radio, contrôle s'il est sélectionné.

Reviewer: ```suggestion
* [`checked`](https://developer.mozilla.org/fr/docs/Web/API/HTMLInputElement#checked) : un booléen. Pour un champ de saisie avec une case à cocher ou un bouton radio, contrôle s'il est sélectionné.
```

Je te laisse remettre les espaces (idéalement insécables, mais je ferai une repasse) sur toutes les props OK ?

Idem pour "bouton de type radio", qui devient "bouton radio" (terminologie très largement ancrée), à remplacer partout en singulier / pluriel.

> * [`value`](https://developer.mozilla.org/fr/docs/Web/API/HTMLInputElement#value): une chaîne de caractères. Pour un champ de saisie textuel, contrôle son texte (pour un bouton de type radio, spécifie la forme des données).

Reviewer: ```suggestion
* [`value`](https://developer.mozilla.org/fr/docs/Web/API/HTMLInputElement#value): une chaîne de caractères. Pour un champ de saisie textuel, contrôle son texte (pour un bouton de type radio, spécifie la valeur soumise avec le formulaire).
```

Glissement sémantique.

Lorsque vous passez une de ces valeurs, vous devez également passer un gestionnaire d'événement `onChange` qui met à jour la valeur passée.

> Ces props de `<input>` sont seulement compatibles avec les champs de saisie non contrôlées :

Reviewer: ```suggestion
Ces props d'`<input>` sont seulement compatibles avec les champs de saisie non contrôlés :
```

- On évite les hiatus avant les blocs de code, donc apostrophe ici.
- C'est le champ qui est contrôlé, pas sa saisie.

> * [`defaultChecked`](https://developer.mozilla.org/fr/docs/Web/API/HTMLInputElement#defaultChecked): un booléen. Spécifie [la valeur initiale](#providing-an-initial-value-for-an-input) pour les champs de saisie `type="checkbox"` et `type="radio"`.

Reviewer: ```suggestion
* [`defaultChecked`](https://developer.mozilla.org/fr/docs/Web/API/HTMLInputElement#defaultChecked): un booléen. Spécifie [la valeur initiale](#providing-an-initial-value-for-an-input) pour les champs de saisie de `type="checkbox"` ou`type="radio"`.
```

En VF passer sur une disjonction est plus naturel ici. À reporter partout ailleurs (y'en a plusieurs, ex. les `formXxxx`).

* [`defaultValue`](https://developer.mozilla.org/fr/docs/Web/API/HTMLInputElement#defaultValue): une chaîne de caractères. Spécifie [la valeur initiale](#providing-an-initial-value-for-an-input) pour un champ de saisie textuel.

> Ces props de `<input>` sont compatibles avec les champs de saisie contrôlés et non contrôlées :

Reviewer: ```suggestion
Ces props d'`<input>` sont compatibles avec les champs de saisie contrôlés comme non contrôlés :
```

> * [`accept`](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input#accept) : une chaîne de caractères. Spécifie quels types de fichiers sont acceptés par un champ de saisie `type="file"`.

Reviewer: ```suggestion
* [`accept`](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input#accept) : une chaîne de caractères. Spécifie quels types de fichiers sont acceptés par un champ de saisie de `type="file"`.
```

Reviewer: Bon OK je te laisse remettre "de " devant les qualificateurs "type=…" un peu partout (ça passait en VO mais en VF l'article manque)

* [`alt`](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input#alt) : une chaîne de caractères. Spécifie le texte alternatif pour un champ de saisie `type="image"`.

> * [`capture`](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input#capture) : une chaîne de caractères. Spécifie le média (microphone, vidéo ou caméra) capturé par un champ de saisie `type="file"`.

Reviewer: Ça me gratte toujours un peu "média" pour un singulier (comme "scénarios" pour un pluriel), mais bon l'usage a consacré ce glissement de la forme officielle latine d'origine (medium, scenarii), alors laissons vivre la langue !

* [`autoComplete`](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input#autocomplete) : une chaîne de caractères. Spécifie l'un des [comportements d'autocomplétion](https://developer.mozilla.org/fr/docs/Web/HTML/Attributes/autocomplete#values) possibles.
* [`autoFocus`](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input#autofocus) : un booléen. Si `true`, React va activer l'élément après le montage _(l'apparition initiale dans le DOM, NdT)_.

> * [`dirname`](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input#dirname) : une chaîne de caractères. Spécifie le nom du champ de saisie pour la directionnalité de l'élément.

Reviewer: ```suggestion
* [`dirname`](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input#dirname) : une chaîne de caractères. Spécifie le nom de la donnée du formulaire indiquant la directionnalité de l'élément.
```

Ce n'est pas un champ (VO pourrie) mais une clé-valeur soumise à l'envoi du formulaire. J'ignorais que cet attribut existait, j'ai dû aller voir le MDN 😉

* [`disabled`](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input#disabled) : un booléen. Si `true`, le champ de saisie ne sera pas interactif et sera grisé.
* `children` : `<input>` n'accepte pas d'enfants.
* [`form`](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input#form) : une chaîne de caractères. Spécifie l'`id` du `<form>` auquel appartient ce champ de saisie. S'il est absent, le champ de saisie sera associé au formulaire parent le plus proche.

> * [`formAction`](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input#formaction) : une chaîne de caractères. Remplace l'attribut `action` du `<form>` parent pour les champs de saisie `type="submit"` et `type="image"`.

Reviewer: ```suggestion
* [`formAction`](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input#formaction) : une chaîne de caractères. Remplace l'attribut `action` du `<form>` parent pour les champs de `type="submit"` ou `type="image"`.
```

On peut difficilement qualifier les champs submit/image de *saisie* : ce sont juste des boutons d'envoi. À reporter ailleurs.

* [`formEnctype`](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input#formenctype) : une chaîne de caractères. Remplace l'attribut `enctype` du `<form>` parent pour les champs de saisie `type="submit"` et `type="image"`.
* [`formMethod`](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input#formmethod) : une chaîne de caractères. Remplace l'attribut `method` du `<form>` parent pour les champs de saisie `type="submit"` et `type="image"`.
* [`formNoValidate`](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input#formnovalidate) : une chaîne de caractères. Remplace l'attribut `novalidate` du `<form>` parent pour les champs de saisie `type="submit"` et `type="image"`.
* [`formTarget`](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input#formtarget) : une chaîne de caractères. Remplace l'attribut `target` du `<form>` parent pour les champs de saisie `type="submit"` et `type="image"`.

> * [`height`](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input#height) : une chaîne de caractères. Spécifie la hauteur de l'image pour `type="image"`.

Reviewer: ```suggestion
* [`height`](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input#height) : une chaîne de caractères. Spécifie la hauteur de l'image pour le `type="image"`.
```

Qualificatif, comme pour "champ de type="… À reporter ailleurs (width sans doute, voire d'autres)

* [`onInvalid`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/invalid_event) : une fonction [gestionnaire d'événement](/reference/react-dom/components/common#event-handler). Se déclenche si un champ de saisie échoue à la validation lors de la soumission du formulaire. Contrairement à l'événement natif `invalid`, l'événement React `onInvalid` se propage.
* `onInvalidCapture` : une version de `onInvalid` qui se déclenche lors de la [phase de capture](/learn/responding-to-events#capture-phase-events).
* [`onSelect`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/select_event) : une fonction [gestionnaire d'événement](/reference/react-dom/components/common#event-handler). Se déclenche après que la sélection à l'intérieur du champ de saisie a changé. React étend l'événement `onSelect` pour se déclencher également pour une sélection vide et sur les modifications de texte (qui peuvent affecter la sélection).
* `onSelectCapture` : une version de `onSelect` qui se déclenche lors de la [phase de capture](/learn/responding-to-events#capture-phase-events).

> * [`pattern`](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input#pattern) : une chaîne de caractères. Spécifie une expression régulière qui doit correspondre à la `value` du champ de saisie.

Reviewer: ```suggestion
* [`pattern`](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input#pattern) : une chaîne de caractères. Spécifie une expression rationnelle qui doit correspondre à la `value` du champ de saisie.
```

Les deux formulations existent, mais "expression rationnelle" est un peu plus répandue — et surtout plus juste sémantiquement. D'ailleurs, c'est ce que le MDN a retenu.

* [`placeholder`](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input#placeholder) : une chaîne de caractères. Affichée dans une couleur discrète lorsque la valeur du champ de saisie est vide.
* [`readOnly`](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input#readonly) : un booléen. Si `true`, le champ de saisie n'est pas modifiable par l'utilisateur.
* [`required`](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input#required) : un booléen. Si `true`, la valeur doit être fournie pour que le formulaire puisse être soumis.

> * [`size`](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input#size) : un nombre. Similaire à la définition de la largeur, mais l'unité dépend du contrôle.

Reviewer: ```suggestion
* [`size`](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input#size) : un nombre. Similaire à la définition de la largeur, mais l'unité dépend de la nature du champ.
```

La VO déconne, mais on dit "field" partout, restons-y.

* [`src`](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input#src) : une chaîne de caractères. Spécifie la source de l'image pour un champ de saisie `type="image"`.

> * [`step`](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input#step) : un nombre positif ou une chaîne `'any'`. Spécifie la distance entre les valeurs valides.

Reviewer: ```suggestion
* [`step`](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input#step) : un nombre positif ou la texte `'any'`. Spécifie la distance entre les valeurs valides.
```

Author: "la texte", c'est normal ? 😁

> * [`type`](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input#type) : une chaîne de caractères. L'un des [types de saisie](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input#input_types).

Reviewer: ```suggestion
* [`type`](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input#type) : une chaîne de caractères. L'un des [types de saisie](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input#les_diff%C3%A9rents_types_de_champs_input).
```

Attention aux liens avec des ancres : la VF a souvent une ancre différente car l'en-tête est traduit.

* [`width`](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input#width) : une chaîne de caractères. Spécifie la largeur de l'image pour un champ de saisie `type="image"`.

#### Limitations {/*caveats*/}

- Les cases à cocher ont besoin de `checked` (ou `defaultChecked`), pas de `value` (ou `defaultValue`).
- Si un champ de saisie reçoit une prop `value` textuelle, il sera [traité comme contrôlé](#controlling-an-input-with-a-state-variable).

> - Si un champ de saisie avec une case à cocher ou un bouton de type radio reçoit une prop `checked` booléenne, il sera [traité comme contrôlé](#controlling-an-input-with-a-state-variable).

Reviewer: ```suggestion
- Si un champ de type case à cocher ou bouton radio reçoit une prop `checked` booléenne, il sera [traité comme contrôlé](#controlling-an-input-with-a-state-variable).
```

> - Un champ de saisie ne peut pas être à la fois contrôlée et non contrôlée.

Reviewer: ```suggestion
- Un champ de saisie ne peut pas être à la fois contrôlé et non contrôlé.
```

Gaffe au copier-coller quand le sujet change de genre ! 😉

- Un champ de saisie ne peut pas basculer entre un statut contrôlé et non contrôlé au cours de son existence.
- Un champ de saisie contrôlé doit avoir un gestionnaire `onChange` qui met à jour sa valeur.

---

## Utilisation {/*usage*/}

### Afficher des champs de saisie de différents types {/*displaying-inputs-of-different-types*/}

> Pour afficher un champ de saisie, affichez un composant `<input>`. Par défaut, il s'agit d'un champ de saisie textuel. Vous pouvez passer `type="checkbox"` pour une case à cocher, `type="radio"` pour un bouton radio, [ou l'un des autres types de saisie](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input#input_types).

Reviewer: ```suggestion
Utilisez un composant `<input>` pour afficher un champ de saisie. Par défaut, il s'agira d'un champ de saisie textuel. Vous pouvez passer `type="checkbox"` pour une case à cocher, `type="radio"` pour un bouton radio, [ou l'un des autres types de saisie](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input#les_diff%C3%A9rents_types_de_champs_input).
```

  return (
    <>
      <label>
        Champ de saisie textuel : <input name="myInput" />
      </label>
      <hr />
      <label>
        Case à cocher : <input type="checkbox" name="myCheckbox" />
      </label>
      <hr />
      <p>

>         Bouton de type radio :

Reviewer: ```suggestion
        Bouton radio :
```

---

### Passer une valeur initiale à un champ de saisie {/*providing-an-initial-value-for-an-input*/}

> Vous pouvez éventuellement spécifier la valeur initiale d'un champ de saisie. Passez-la comme une chaîne de caractères pour les champs de saisie textuels. Les cases à cocher et les boutons radio doivent spécifier la valeur initiale avec le booléen `defaultChecked`.

Reviewer: ```suggestion
Vous pouvez éventuellement spécifier la valeur initiale d'un champ de saisie. Passez-la comme une chaîne de caractères `defaultValue` pour les champs de saisie textuels. Les cases à cocher et les boutons radio spécifient quant à eux la valeur initiale avec le booléen `defaultChecked`.
```

  return (
    <>
      <label>

>         Champ de saisie textuel : <input name="myInput" defaultValue="Some initial value" />

Reviewer: ```suggestion
        Champ de saisie textuel : <input name="myInput" defaultValue="Une valeur initiale" />
```

      </label>
      <hr />
      <label>
        Case à cocher : <input type="checkbox" name="myCheckbox" defaultChecked={true} />
      </label>
      <hr />
      <p>

>         Bouton de type radio :

Reviewer: ```suggestion
        Bouton radio :
```

<Sandpack>

```js
export default function MyForm() {
  function handleSubmit(e) {
    // Empêche le navigateur de recharger la page
    e.preventDefault();

    // Lit les données du formulaire
    const form = e.target;
    const formData = new FormData(form);

    // Vous pouvez passer formData directement comme
    // corps de la requête à fetch:
    fetch('/some-api', { method: form.method, body: formData });

    // Ou vous pouvez travailler avec comme un
    // objet simple :
    const formJson = Object.fromEntries(formData.entries());
    console.log(formJson);
  }

  return (
    <form method="post" onSubmit={handleSubmit}>
      <label>

>         Champ de saisie textuel : <input name="myInput" defaultValue="Some initial value" />

Reviewer: ```suggestion
        Champ de saisie textuel : <input name="myInput" defaultValue="Une valeur initiale" />
```

      </label>
      <hr />
      <label>
        Case à cocher : <input type="checkbox" name="myCheckbox" defaultChecked={true} />
      </label>
      <hr />
      <p>

>         Bouton de type radio :

Reviewer: ```suggestion
        Bouton radio :
```

<Note>

Donnez un `name` à votre `<input>`, par exemple `<input name="firstName" defaultValue="Clara" />`. Le `name` que vous avez spécifié sera utilisé comme clé dans les données du formulaire, par exemple `{ firstName: "Clara" }`.

</Note>

<Pitfall>

> Par défaut, *n'importe quel* `<button>` à l'intérieur d'un `<form>` va le soumettre. Cela peut être surprenant ! Si vous avez votre propre composant React `Button`, envisagez de renvoyer [`<button type="button">`](https://developer.mozilla.org/fr/docs/Web/HTML/Element/input/button) au lieu de `<button>`. Ensuite, pour être explicite, utilisez `<button type="submit">` pour les boutons qui *sont* censés soumettre le formulaire.

Reviewer: Ça me fait toujours autant tiquer ce truc, faut que je leur ouvre une *issue* dans la VO…

</Pitfall>

---

### Contrôler un champ de saisie avec une variable d'état {/*controlling-an-input-with-a-state-variable*/}

> Un champ de saisie comme `<input />` est *non contrôlée*. Même si vous [passez une valeur initiale](#providing-an-initial-value-for-an-inpuy) comme `<input defaultValue="Texte initial" />`, votre JSX ne spécifie que la valeur initiale, il ne contrôle pas la valeur actuelle.

Reviewer: ```suggestion
Un champ de saisie comme `<input />` est *non contrôlé*. Même si vous [passez une valeur initiale](#providing-an-initial-value-for-an-input) comme `<input defaultValue="Texte initial" />`, votre JSX ne spécifie que la valeur initiale, il ne contrôle pas la valeur actuelle.
```

Subtil, la typo dans la trad de l'ancre 😉

Author: Wow, je sais pas comment je me suis débrouillée pour celle-ci ! 😅 Bien vu !

<Pitfall>

**Si vous passez `value` sans `onChange`, il sera impossible de réaliser une saisie dans le champ de saisie.** Lorsque vous contrôlez un champ de saisie en passant une `value`, vous *forcez* le champ de saisie à toujours avoir la valeur que vous avez passée. Donc, si vous passez une variable d'état comme `value` mais oubliez de mettre à jour cette variable d'état de manière synchrone au sein du gestionnaire d'événement `onChange`, React réinitialisera le champ de saisie, après chaque frappe, à la `value` que vous avez spécifiée.

</Pitfall>

---

> ### Optimiser le ré-affichage à chaque frappe sur le clavier {/*optimizing-re-rendering-on-every-keystroke*/}

Reviewer: ```suggestion
### Optimiser le recalcul de rendu à chaque frappe clavier {/*optimizing-re-rendering-on-every-keystroke*/}
```

> Lorsque vous utilisez un champ de saisie contrôlé, vous définissez l'état à chaque frappe. Si le composant contenant votre état affiche de nouveau un grand arbre, ça peut devenir lent. Il existe plusieurs façons d'optimiser les performances de ré-affichage.

Reviewer: ```suggestion
Lorsque vous utilisez un champ de saisie contrôlé, vous modifiez l'état à chaque frappe. Si le composant contenant votre état recalcule le rendu d'un grand arbre, ça peut devenir lent. Il existe plusieurs façons d'optimiser les performances d'un recalcul de rendu.
```

On n'est pas tant sur un ré-affichage (qui peut être assez rapide) que sur la phase de rendu elle-même, et ici le distingo compte.

> Par exemple, supposons que vous commencez avec un formulaire qui ré-affiche tout le contenu de la page à chaque frappe :

Reviewer: ```suggestion
Par exemple, supposons que vous commenciez avec un formulaire qui recalcule tout le contenu de la page à chaque frappe :
```

"Supposer" / "imaginer" + subjonctif (car non avéré). "Dire" + indicatif (car avéré).

}
```

> Ça améliore considérablement les performances car maintenant seul `SignupForm` se ré-affiche à chaque frappe.

Reviewer: ```suggestion
Ça améliore considérablement les performances car maintenant seul `SignupForm` se recalcule à chaque frappe.
```

> S'il n'y a aucun moyen d'éviter le ré-affichage (par exemple, si `PageContent` dépend de la valeur de la saisie de recherche), [`useDeferredValue`](/reference/react/useDeferredValue#deferring-re-rendering-for-a-part-of-the-ui) vous permet de garder le champ de saisie contrôlé réactif même au milieu d'un grand ré-affichage.

Reviewer: ```suggestion
S'il n'y a aucun moyen d'éviter le recalcul (par exemple, si `PageContent` dépend de la valeur de la saisie de recherche), [`useDeferredValue`](/reference/react/useDeferredValue#deferring-re-rendering-for-a-part-of-the-ui) vous permet de conserver un champ de saisie contrôlé réactif même au sein d'un recalcul coûteux.
```

</ConsoleBlock>

> Comme le message d'erreur le suggère, si vous ne voulez spécifier que [la valeur *initiale*](#providing-an-initial-value-for-an-input) de la case à cocher, passez plutôt `defaultValue` :

Reviewer: ```suggestion
Comme le message d'erreur le suggère, si vous ne voulez spécifier que [la valeur *initiale*](#providing-an-initial-value-for-an-input) de la case à cocher, passez plutôt `defaultChecked` :
```

Le copier-coller, ce meilleur ami / pire ennemi.

```js
// ✅ Correct : case à cocher non contrôlée avec une valeur initiale
<input type="checkbox" defaultChecked={something} />
```

Si vous voulez [contrôler cette case à cocher avec une variable d'état](#controlling-an-input-with-a-state-variable), spécifiez un gestionnaire `onChange` :

```js
// ✅ Correct : case à cocher contrôlée avec onChange
<input type="checkbox" checked={something} onChange={e => setSomething(e.target.checked)} />
```

<Pitfall>

Vous devez lire `e.target.checked` plutôt que `e.target.value` pour les cases à cocher.

</Pitfall>

Si la case à cocher est en lecture seule intentionnellement, ajoutez une prop `readOnly` pour supprimer l'erreur :

```js
// ✅ Correct : champ de saisie en lecture seule sans onChange
<input type="checkbox" checked={something} readOnly={true} />
```

---

### Le curseur de mon champ de saisie revient au début à chaque frappe {/*my-input-caret-jumps-to-the-beginning-on-every-keystroke*/}

Si vous [contrôlez un champ de saisie](#controlling-an-input-with-a-state-variable), vous devez mettre à jour sa variable d'état avec la valeur DOM du champ de saisie pendant le `onChange`.

> Vous ne pouvez pas la mettre à jour avec autre chose que `e.target.value` (ou `e.target.checked` pour les cases à cocher) : 

Reviewer: Là aussi j'attends leur retour sur l'issue ouverte, car c'est factuellement faux.

> Vous ne pouvez pas la mettre à jour avec autre chose que `e.target.value` (ou `e.target.checked` pour les cases à cocher) : 

Reviewer: ```suggestion
Vous ne pouvez pas le mettre à jour avec autre chose que `e.target.value` (ou `e.target.checked` pour les cases à cocher) : 
```

```js
function handleChange(e) {
  // 🔴 Bug : mettre à jour un champ de saisie avec autre chose que e.target.value
  setFirstName(e.target.value.toUpperCase());
}
```

> Vous ne pouvez pas la mettre à jour de manière asynchrone :

Reviewer: ```suggestion
Vous ne pouvez pas le mettre à jour de manière asynchrone :
```
